ufo/ufo-scheduler.c

```diff
@@ -321,14 +321,20 @@ check_target_connections (UfoTaskGraph *graph,
         label = ufo_graph_get_edge_label (UFO_GRAPH (graph),
                                           UFO_NODE (it->data), target);
         input = GPOINTER_TO_INT (label);
-        g_assert (input >= 0 && input < 16);
-        connection_bitmap |= 1 << input;
+        g_assert (input >= 0 && input < UFO_MAX_INPUT_NODES);
+        connections[input] = TRUE;
     }
 
-    mask = (1 << n_inputs) - 1;
+
+
+    gbool all_inputs_connected = TRUE;
+    for(int i = 0; i < n_inputs; ++i){
+        if(connections[input] == FALSE)
+            all_inputs_connected = FALSE;
```

Comment:
```suggestion
        all_inputs_connected &= connections[input];
```
seems to express better what this loop does. By the way: three newlines seem a bit excessive ;-)